src/pytorch_lightning/strategies/fsdp.py

```diff
@@ -71,6 +71,20 @@
 log = logging.getLogger(__name__)
 
 
+class _FSDPStrategyModuleWrapper(_LightningModuleWrapperBase):
+    def state_dict(self, *args: Any, **kwargs: Any) -> Dict[str, Any]:  # type: ignore[override]
+        # this is required because with FSDP lightning_module is empty because weights are sharded.
+        # So we need to call self.trainer.model.state_dict (wrapped version) and use this wraper to
+        # avoid extra keys `_forward_module.layer.weight.` since we want `layer.weight.` in state_dict.
+        return self._forward_module.state_dict(*args, **kwargs)
+
+    def named_modules(self, *args: Any, **kwargs: Any) -> Iterator[Tuple[str, Module]]:
```

Comment:
I wonder if it would be useful to put this in the parent (`_LightningModuleWrapperBase`)

Reply:
Are there any other strategies or another classes that already somehow handles `_forward_module`? If yes, then directly changing `_LightningModuleWrapperBase` would break them.